BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/Entry.groovy

```diff
@@ -0,0 +1,37 @@
+package com.novoda.buildproperties
+
+class Entry {
+
+    private final String key
+    private final Closure<Object> value
+
+    Entry(String key, Closure<Object> value) {
+        this.key = key
+        this.value = value
+    }
+
+    String getKey() {
+        key
+    }
+
+    Boolean getBoolean() {
+        Boolean.parseBoolean(string)
+    }
+
+    Integer getInt() {
+        getValue() as Integer
+    }
+
+    Double getDouble() {
+        getValue() as Double
+    }
+
+    String getString() {
+        getValue() as String
+    }
+
+    Object getValue() {
+        value.call()
```

Comment:
In a future `or` implementation, this is where you'd want to catch something like `CompositeException` and, if `value.call()` fails, you add the new exception to the previously thrown `CompositeException`.
This way you have an exception that says something like 'Could not find values for keys: a, b, c, d.'.